src/cascadia/TerminalSettingsModel/AppearanceConfig.cpp

```diff
@@ -172,20 +172,20 @@ winrt::hstring AppearanceConfig::ExpandedBackgroundImagePath()
     }
 }
 
-void AppearanceConfig::_logSettingSet(std::string_view setting)
+void AppearanceConfig::_logSettingSet(const std::string& setting)
 {
     _changeLog.emplace(setting);
 }
 
-void AppearanceConfig::_logSettingIfSet(std::string_view setting, const bool isSet)
+void AppearanceConfig::_logSettingIfSet(const std::string_view& setting, const bool isSet)
 {
     if (isSet)
     {
-        _logSettingSet(setting);
+        _logSettingSet(std::string{ setting });

```

Comment:
Okay, so here's what I would have done.

`_logSettingSet` doesn't need to take a `string`. that reduces its flexibility and forces all callers (like R113 above) to KNOW that it needs a string.

It can still take a string_view, and it *should* take a string_view. It can do the conversion to `string` at the last possible moment, right when it inserts it into the set.

You don't need to let the set's implementation details leak out to any callers who don't know about the set!